Sources/MQTTNIO/MQTTConnection.swift

```diff
@@ -75,7 +74,16 @@ final class MQTTConnection {
                         return channel.pipeline.addHandlers(handlers)
                     }
                 }
-                .connect(host: client.host, port: client.port)
+
+            let channelFuture: EventLoopFuture<Channel>
+
+            if client.port == 0 {
```

Comment:
Not a huge fan of the assume it is a unix domain path if port is zero, but yeah can't do this without a breaking change. When this gets merged we should add an issue to clean this up for the next major release

Reply:
Agreed that this approach feels a bit icky.

We could have a new/alternative `init` that takes a `unixSockePath` instead of `host` and `port` parameters. I think this would still maintain compatibility and would be a better interface for initializing an `MQTTClient`, but since `MQTTClient.host` and `MQTTClient.port` are both public attributes, we could not fully hide this from the user.

Reply:
I've updated the PR with a change that adds a new convenience init for `MQTTClient` that takes a `unixSocketPath` parameter instead of `host` and `port`. This feels like an improvement and seems fine from a compatibility standpoint.